synapse/http/connectproxyclient.py

```diff
@@ -59,7 +59,7 @@ def as_proxy_authorization_value(self) -> bytes:
             a Proxy-Authorization header.
         """
         # Encode as base64 and prepend the authorization type
-        return b"Basic " + base64.encodebytes(self.username_password)
+        return b"Basic " + base64.encodebytes(self.username_password).replace(b'\n', b'')
```

Comment:
When does this end up with line breaks in it?

Reply:
> from [docs.python.org/3/library/base64.html](https://docs.python.org/3/library/base64.html)

Specifically https://docs.python.org/3/library/base64.html#base64.encode

Reply:
I wonder if we ought to be using https://docs.python.org/3/library/base64.html#base64.b64encode?

Reply:
I think I've always used `base64.b64encode` in the past.

Reply:
ill push that then

Reply:
```python
>>> base64.encodebytes(f"l{'o'*65}nguser:pass".encode())
b'bG9vb29vb29vb29vb29vb29vb29vb29vb29vb29vb29vb29vb29vb29vb29vb29vb29vb29vb29v\nb29vb29vb29vbmd1c2VyOnBhc3M=\n'
>>> base64.b64encode(f"l{'o'*65}nguser:pass".encode())
b'bG9vb29vb29vb29vb29vb29vb29vb29vb29vb29vb29vb29vb29vb29vb29vb29vb29vb29vb29vb29vb29vb29vbmd1c2VyOnBhc3M='
```

Synapse doesn't use encodebytes anywhere else. Let's use b64encode!

Reply:
> I've never in my life seen such behavior but apparently it's a thing 🤷‍♂️

One person speculates that this was useful when sending binary files in email and USENET bodies, see e.g. https://superuser.com/a/1225139/550042